File: 백준/Silver/1193. 분수찾기/분수찾기.py
x = int(input())
line = 1

while line < x:
    x -= line
    line += 1

# 홀수 번째 라인일 경우
if line % 2:
    print(f'{line - x + 1}/{x}')
# 짝수 번째 라인일 경우
else:
    print(f'{x}/{line - x + 1}')

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int line = 1;
        int X = sc.nextInt();
        
        while (X > line) {
            X -= line;
            line++;
        }

        if (line % 2 == 0) System.out.println(X + "/" + (line - X + 1));
        else System.out.println((line - X + 1) + "/" + X);
    }
}